File: src/WorkBC.Shared/Services/SmtpEmailSender.cs
using System;
using System.Net.Mail;
using System.Net.Mime;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Extensions.Options;
using WorkBC.Shared.Settings;

namespace WorkBC.Shared.Services
{
    public class SmtpEmailSender : IEmailSender
    {
        private readonly EmailSettings _emailSettings;
        private readonly Serilog.ILogger _logger;

        public SmtpEmailSender(IOptions<EmailSettings> emailSettings, Serilog.ILogger logger)
        {
            _emailSettings = emailSettings.Value;
            _logger = logger;
        }

        public SmtpEmailSender(EmailSettings emailSettings, Serilog.ILogger logger)
        {
            _emailSettings = emailSettings;
            _logger = logger;
        }

        public async Task SendEmailAsync(string email, string subject, string htmlMessage, string textMessage)
        {
            if (string.IsNullOrWhiteSpace(email) || string.IsNullOrWhiteSpace(subject))
            {
                return;
            }

            string fromEmail = _emailSettings.FromEmail ?? "noreply@gov.bc.ca";
            string fromName = _emailSettings.FromName ?? "WorkBC.ca No Reply";

            try
            {
                var client = new SmtpClient(_emailSettings.SmtpServer, _emailSettings.SmtpPort);

                var mailMessage = new MailMessage
                {
                    From = new MailAddress(fromEmail, fromName),
                    Body = textMessage,
                    Subject = subject
                };

                mailMessage.AlternateViews.Add(
                    AlternateView.CreateAlternateViewFromString(htmlMessage, Encoding.UTF8, "text/html"));

                mailMessage.To.Add(email);

                await client.SendMailAsync(mailMessage);
            }
            catch (Exception ex)
            {
                _logger.Error(ex, "SmtpEmailSender.SendEmailAsync failed");
                throw;
            }
        }
    }
}

File: src/WorkBC.Notifications.JobAlerts.V2/src/Email/SmtpEmailSender.php
<?php

declare(strict_types=1);

namespace WorkBC\JobAlertNotifier\Email;

use Monolog\Logger;
use PHPMailer\PHPMailer\PHPMailer;
use WorkBC\JobAlertNotifier\Config\AppConfig;

/**
 * Port of WorkBC.Shared.Services.SmtpEmailSender.
 *
 * Plain (unauthenticated, non-TLS) SMTP relay with a text body and an HTML
 * alternate view, matching the C# System.Net.Mail.SmtpClient usage.
 */
final class SmtpEmailSender implements EmailSenderInterface
{
    public function __construct(
        private readonly AppConfig $config,
        private readonly Logger $log,
    ) {
    }

    public function sendEmail(string $email, string $subject, string $htmlMessage, string $textMessage): void
    {
        if (trim($email) === '' || trim($subject) === '') {
            return;
        }

        $fromEmail = $this->config->emailFromEmail !== '' ? $this->config->emailFromEmail : 'noreply@gov.bc.ca';
        $fromName = $this->config->emailFromName !== '' ? $this->config->emailFromName : 'WorkBC.ca No Reply';

        try {
            $mailer = new PHPMailer(true);
            $mailer->isSMTP();
            $mailer->Host = $this->config->emailSmtpServer;
            $mailer->Port = $this->config->emailSmtpPort;
            $mailer->SMTPAuth = false;
            $mailer->SMTPAutoTLS = false;
            $mailer->CharSet = PHPMailer::CHARSET_UTF8;

            $mailer->setFrom($fromEmail, $fromName);
            $mailer->addAddress($email);
            $mailer->Subject = $subject;
            $mailer->isHTML(true);
            $mailer->Body = $htmlMessage;
            $mailer->AltBody = $textMessage;

            $mailer->send();
        } catch (\Throwable $e) {
            $this->log->error('SmtpEmailSender::sendEmail failed: ' . $e->getMessage());
            throw $e;
        }
    }
}
